account_tax_expense_include/hr_expense_expense.py

```diff
@@ -0,0 +1,127 @@
+
+from openerp.osv import orm
+
+
+class hr_expense_expense(orm.Model):
+    _inherit = 'hr.expense.expense'
+
+    def move_line_get(self, cr, uid, expense_id, context=None):
+        res = []
+        tax_obj = self.pool.get('account.tax')
+        cur_obj = self.pool.get('res.currency')
+        if context is None:
+            context = {}
+        exp = self.browse(cr, uid, expense_id, context=context)
+        company_currency = exp.company_id.currency_id.id
+
+        for line in exp.line_ids:
+            mres = self.move_line_get_item(cr, uid, line, context)
+            if not mres:
+                continue
+            res.append(mres)
+            tax_code_found = False
+
+            # Calculate tax according to default tax on product
+            taxes = []
+            # Taken from product_id_onchange in account.invoice
+            if line.product_id:
+                fpos_obj = self.pool.get('account.fiscal.position')
+
+                product = line.product_id
+                taxes = product.supplier_taxes_id
+                fpos = line.partner_id.property_account_position
+
+                # If taxes are not related to the product, maybe they are in
+                # the account
+                if not taxes:
+                    expense_account = product.property_account_expense
+                    a = expense_account and expense_account.id
+
+                    if not a:
+                        expense_account = product.categ_id.\
+                            property_account_expense
+                        a = expense_account and expense_account.id
+
+                    a = fpos_obj.map_account(cr, uid, fpos, a)
+
+                    taxes = a and self.pool.get('account.account').browse(
+                        cr, uid, a, context=context).tax_ids or False
+
+                tax_ids = fpos_obj.map_tax(cr, uid, fpos, taxes)
+                taxes = tax_obj.browse(
+                    cr, uid, tax_ids, context=context) or taxes
+
+            if not taxes:
+                continue
+
+            src_move_line = res[-1]
+            # Calculating tax on the line and creating move?
+            for tax in tax_obj.compute_all(cr, uid, taxes,
+                                           line.unit_amount,
+                                           line.unit_quantity, line.product_id,
+                                           exp.user_id.partner_id,
+                                           is_expense=True)['taxes']:
+
+                tax_code_id = tax['base_code_id']
+                tax_amount = line.total_amount * tax['base_sign']
+                if tax_code_found:
+                    if not tax_code_id:
+                        continue
+                    res.append(self.move_line_get_item(cr, uid, line, context))
+                    res[-1]['price'] = 0.0
+                    res[-1]['account_analytic_id'] = False
+                elif not tax_code_id:
+                    continue
```

Comment:
If you put `if not tax_code_id` first and `elif tax_code_found`, you won't have to check `tax_code_id` twice